protocols/identify/src/behaviour.rs

```diff
@@ -400,30 +377,11 @@ impl NetworkBehaviour for Behaviour {
             }) => {
                 if remaining_established == 0 {
                     self.connected.remove(&peer_id);
-                    self.requests.retain(|request| {
-                        request
-                            != &Request {
-                                peer_id,
-                                protocol: Protocol::Push,
-                            }
-                    });
```

Comment:
I think this is fine because we drop messages to non-connected peers anyway.

Reply:
:+1: especially since `requests` is now gone entirely.

Reply:
I did wonder at some point whether that is a good thing. In a way, I think `rust-libp2p` does a half-finished job if it provides me a message passing abstraction but then just silently drops them.

I wonder if we shouldn't at least print a warning or give the command back to the behaviour.